### DISCUSSION TITLE
User specific path possible?

### DISCUSSION THREAD

**Discussion by roelbroersma** [r7-labs/R7.Documents#119] [General]
Hello,

Instead of having the documents module on 1 path, i.e:  /<siteroot>/documents/

I would like to have the documents module in a user specific path like:  /<siteroot>/documents/<userID>/

Is it possible to use a token in here?    Or modify something in the database or Codebehind file somewhere?   The module:  "Document Exchange"   from Peter Donker  (www.bring2mind.net) has such an option.

Kind regards,

Roel Broersma

**Comment by roman-yagodin** [comment]
I don't think I understand - what kind of "path" do you mean?

**Comment by roelbroersma** [comment]
I mean the document path.   Where uploaded files are stored.

**Comment by roman-yagodin** [comment]
The R7.Documents module doesn't controls where files are stored - it could be any existing path in the filesystem which is available (accessible) to the user.

**Comment by roman-yagodin** [comment]
Upload functionality currently provided via "standard" DnnUrlControl - which is not very configurable (nor reliable) thing. I really want to replace it with something more sane in all my DNN projects...

It also possible to use Digital Assets Management (DAM) module to setup upload locations. It can be configured to show only current user folder, or group (role) folder, or any specific folder.

**Comment by roelbroersma** [comment]
DAM module still uses old (deprecated) Telerik upload controls.  See:  https://dnnsupport.dnnsoftware.com/hc/en-us/articles/360004928673-Using-the-Digital-Asset-Management-DAM-module-with-EVOQ-9-0-and-above

I think it will soon be removed from DNN at all...

**Comment by roelbroersma** [comment]
BTW.  Do you understand what i mean?

Example:
Have a DNN website with about 100 users/accounts.

When the users login they can go to a page on which the R7.Documents module is loaded.   There they can upload files to their own folder,  example:  /<siteroot>/Portal/0/users/<user_id>/

They can not navigate outside this area to a higher level.  So they can not reach other users folders.

**Comment by roman-yagodin** [comment]
I think I understand now what you mean, but really have no plans to create some kind of self-contained solution from R7.Documents - I'd pretty much like to keep it KISS and deeply integrated into the DNN Platform, reusing and relying on as much platform functionality as possible. This approach is currently suits well for my use cases. Though you are right, missing upload functionality could be a weak point.

By the way, DAM module cannot be just removed - it's an essential part of DNN, so it can only be  replaced with something with similar functionality, or it will be a major regression. Web CMS without file manager?! Give me a break... And in your case you need only 1 page with 1 DAM module instance configured to user folder mode.
